pilot/pkg/proxy/envoy/v2/discovery.go

```diff
@@ -183,6 +187,16 @@ func NewDiscoveryServer(
 		return nil
 	}
 
+	instanceUpdateHandler := func(instance *model.ServiceInstance, event model.Event) {
+		err := out.updateProxyServiceInstances(instance)
```

Comment:
seems this will be called for every ip:port even though we only really care about IPs (I assume kubernetes will always created all ports at the same time as IPs but maybe not?). Shouldn't be a big issue though

Reply:
Right, will be called for every ip:port resolved by the endpoint's subsets. However, it'll quickly realize there's no change in the proxy's service instances, and will therefore avoid a redundant push.